workitem/workitem_repository.go

```diff
@@ -110,35 +110,31 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 		return nil, errors.NewBadParameterError("Type", wi.Type)
 	}
 
-	newWi := WorkItem{
-		ID:      id,
-		Type:    wi.Type,
-		Version: wi.Version + 1,
-		Fields:  Fields{},
-	}
+	res.Version = res.Version + 1
+	res.Type = wi.Type
```

Comment:
I think we do not need this Type assignment. (Can be skipped). Because we are not allowing type change yet.

Also, Fields are not reset. IMO we need to reset ?
While update API, we replace whole set of `Fields` with new input `Fields` but looking at this change we will be keeping `old values in Fields`.